contracts/voting/dao-voting-cw4/src/contract.rs

```diff
@@ -108,12 +130,70 @@ pub fn instantiate(
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn execute(
-    _deps: DepsMut,
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::UpdateActiveThreshold { new_threshold } => {
+            execute_update_active_threshold(deps, env, info, new_threshold)
+        }
+    }
+}
+
+pub fn execute_update_active_threshold(
+    deps: DepsMut,
     _env: Env,
-    _info: MessageInfo,
-    _msg: ExecuteMsg,
+    info: MessageInfo,
+    new_active_threshold: Option<ActiveThreshold>,
 ) -> Result<Response, ContractError> {
-    Err(ContractError::NoExecute {})
+    let dao = DAO.load(deps.storage)?;
+    if info.sender != dao {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    if let Some(active_threshold) = &new_active_threshold {
+        // Pre-validation before state changes
+        match active_threshold {
+            ActiveThreshold::Percentage { percent } => {
+                assert_valid_percentage_threshold(*percent)?;
+            }
```

Comment:
I think we should not allow `ActiveThreshold::Percentage` for this contract, so we should throw an error here.